optuna/pruners/hyperband.py

```diff
@@ -49,18 +47,37 @@ class HyperbandPruner(BasePruner):
         Thus, for example, if ``HyperbandPruner`` has :math:`4` pruners in it,
         at least :math:`4 \\times 10` pruners are consumed for startup.
 
+    .. note::
+        Hyperband has several :class:`~optuna.pruners.SuccessiveHalvingPruner`. Each
+        :class:`~optuna.pruners.SuccessiveHalvingPruner` is referred as "bracket" in the original
+        paper. The number of brackets is an important factor to control the early stopping behavior
+        of Hyperband and is determined by ``max_resource`` and ``reduction_factor`` as
+        `n_brackets = floor(log(max_resource) / log(reduction_factor))`. Please set
+        ``max_resource`` and ``reduction_factor`` so that the number of brackets is not too large
+        (about 4 ~ 10 in most use cases).
```

Comment:
IIRC, the paper recommends 3 and 4.

Reply:
Thank you for your comment. Please see Section 3.6 of the [original paper](https://arxiv.org/pdf/1603.06560.pdf). In the first paragraph of Section 3.6, we should set `\eta` to 3 or 4. In the second paragraph, it is noted how many brackets should be used. It is said that the 5 brackets are good to take `n` versus `B/n` tradeoff. I change the comment as "about 4 ~ 6 in most use cases". Thank you for your insightful remarks.